src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.cs

```diff
@@ -248,6 +247,144 @@ public static void FaultException_Throws_WithFaultDetail()
+                    error += String.Format("\r\nInnerException:\r\n{0}", e.InnerException.ToString());
+                errorBuilder.AppendLine(error);
+            }
+            else
+            {
+                FaultException<FaultDetail> faultException = (FaultException<FaultDetail>)(e);
+                string actualFaultMsg = ((FaultDetail)(faultException.Detail)).Message;
+                if (actualFaultMsg != faultMsg)
+                {
+                    errorBuilder.AppendLine(string.Format("Expected Fault Message: {0}, actual: {1}", faultMsg, actualFaultMsg));
+                }
+            }
+
+            Assert.True(errorBuilder.Length == 0, string.Format("Test Scenario: FaultException_Throws_WithFaultDetail FAILED with the following errors: {0}", errorBuilder));
+            return;
+        }
+
+        Assert.True(false, "Expected FaultException<FaultDetail> exception, but no exception thrown.");
+    }
+
+    [Theory]
+    [OuterLoop]
+    public static void FaultException_Throws_FaultDetail2()
+    {
+        string faultMsg = "Test Fault Exception";
+        StringBuilder errorBuilder = new StringBuilder();
+
+        try
+        {
+            BasicHttpBinding binding = new BasicHttpBinding();
+            using (ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic)))
+            {
+                IWcfService serviceProxy = factory.CreateChannel();
+                serviceProxy.TestFaults(faultMsg, false);
+            }
+        }
+        catch (Exception e)
+        {
+            if (e.GetType() != typeof(FaultException<FaultDetail2>))
+            {
+                string error = string.Format("Expected exception: {0}, actual: {1}\r\n{2}",
+                                             "FaultException<FaultDetail2>", e.GetType(), e.ToString());
+                if (e.InnerException != null)
+                    error += String.Format("\r\nInnerException:\r\n{0}", e.InnerException.ToString());
+                errorBuilder.AppendLine(error);
+            }
+            else
+            {
+                FaultException<FaultDetail2> faultException = (FaultException<FaultDetail2>)(e);
+                string actualFaultMsg = ((FaultDetail2)(faultException.Detail)).Message;
+                if (actualFaultMsg != faultMsg)
+                {
+                    errorBuilder.AppendLine(string.Format("Expected Fault Message: {0}, actual: {1}", faultMsg, actualFaultMsg));
+                }
+            }
+
+            Assert.True(errorBuilder.Length == 0, string.Format("Test Scenario: FaultException_Throws_WithFaultDetail2 FAILED with the following errors: {0}", errorBuilder));
+            return;
+        }
+
+        Assert.True(false, "Expected FaultException<FaultDetail2> exception, but no exception thrown.");
+    }
+
+    [Theory]
+    [OuterLoop]
+    public static void FaultException_Echo_Or_Throws_Fault_Exception()
```

Comment:
I think these tests probably belong in Contract\Fault\FaultExceptionTests.cs which is dedicated to testing fault exceptions.  And as I mentioned below, this test should be broken into 2 tests -- one that tests the non-exception path, and the other that tests the exception path.

Incidentally -- what is the purpose of the exception path test?  Does it cover any new scenario?  We already have tests here to validate FaultDetail gets thrown.  It looks like the TestFaultWithKnownType is really intended to be used to round-trip an array of known types.  And this specific test appears to be that a [FaultContract] becomes a known type because it is part of the contract.  If that's really what it tests, then the positive test should probably be something like "FaultDetail2_RoundTrips_As_Known_Type()".